ivy/data_classes/container/experimental/losses.py

```diff
@@ -0,0 +1,169 @@
+        >>> target = ivy.Container(a=ivy.array(1),
+        >>> b=ivy.array(-1))
+        >>> loss = ivy.Container.static_margin_ranking_loss(true, pred, target)
+        >>> print(loss)
+        {
+            a: ivy.array(0.2),
+            b: ivy.array(0.4)
+        }
+        """
+        return ContainerBase.cont_multi_map_in_function(
+            "margin_ranking_loss",
+            true,
+            pred,
+            target,
+            margin=margin,
+            reduction=reduction,
+            key_chains=key_chains,
+            to_apply=to_apply,
+            prune_unapplied=prune_unapplied,
+            map_sequences=map_sequences,
+            out=out,
+        )
+
+    def margin_ranking_loss(
+        self: ivy.Container,
+        pred: Union[ivy.Container, ivy.Array, ivy.NativeArray],
+        target: Union[ivy.Container, ivy.Array, ivy.NativeArray],
+        /,
+        *,
+        margin: Union[float, ivy.Container] = 0.0,
+        reduction: Union[str, ivy.Container] = "mean",
+        key_chains: Optional[Union[List[str], Dict[str, str], ivy.Container]] = None,
+        to_apply: Union[bool, ivy.Container] = True,
+        prune_unapplied: Union[bool, ivy.Container] = False,
+        map_sequences: Union[bool, ivy.Container] = False,
+        out: Optional[ivy.Container] = None,
+    ) -> ivy.Container:
+        """
+        ivy.Container instance method variant of ivy.margin_ranking_loss. This method
+        simply wraps the function, and so the docstring for ivy.margin_ranking_loss also
+        applies to this method with minimal changes.
+
+        Parameters
+        ----------
+        self
+            input container containing predictions for the first input.
+        true
+            input array or container containing predictions for the first input.
+        pred
+            input array or container containing predictions for the second input.
+        target
+            input array or container containing the binary labels (1 or -1).
+        margin
+            margin for the loss. Default is ``0``.
+        reduction
+            Specifies the reduction to apply to the output. Default is ``"mean"``.
+        key_chains
+            The key-chains to apply or not apply the method to. Default is ``None``.
+        to_apply
+            If True, the method will be applied to key_chains, otherwise key_chains
+            will be skipped. Default is ``True``.
+        prune_unapplied
+            Whether to prune key_chains for which the function was not applied.
+            Default is ``False``.
+        map_sequences
+            Whether to also map method to sequences (lists, tuples).
+            Default is ``False``.
+        out
+            optional output container, for writing the result to. It must have a shape
+            that the inputs broadcast to.
+
+        Returns
+        -------
+        ret
+            The margin ranking loss.
+
+        Examples
```

Comment:
What is going on in this example?

Reply:
@tomatillos The code provides examples of how to use the methods defined in the class. In both examples, they show how to create input Container instances, where each container has multiple named sub-containers. These sub-containers seem to correspond to different data samples or data points. The examples then demonstrate how to calculate the margin ranking loss for the provided inputs.